### ISSUE TITLE
Type checking not working on empty class/interface

### ISSUE THREAD

**Issue by Lusito** [Microsoft/TypeScript#19408]
**TypeScript Version:**   typescript@2.7.0-dev.20171021

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
class SomeBaseClass {}
let c: SomeBaseClass = 13;

```

**Expected behavior:**
The last line should show an error, since the value is of type number, not SomeBaseClass.

**Actual behavior:**
No error or warning shows. As soon as SomeBaseClass gets attributes or methods (aside from the constructor), an error is actually shown.

**Comment by aluanhaddad** [comment]
Typescript has a structural type system. `SomeBaseClass` is a Constructor that creates objects that have no properties, thier being `{}`, a constraint which numbers satisfy. Do not create empty classes, interfaces or types.

**Comment by Lusito** [comment]
I use SomeBaseClass as parent class, so I can pass extended classes to that test method. I think that is a valid use-case.

**Comment by Lusito** [comment]
ok, it seems this has nothing to do with generics.. I've updated the bug description and example code to be even simpler

**Comment by jcalz** [comment]
As @aluanhaddad said, you want nominal typing, which TypeScript doesn't really have.  This is a duplicate of #202; instead of discussing here you might want to go to #202 and give it a 👍, and if you think you have a new use case you can mention it there (although nominal classes is probably well covered).

**Comment by RyanCavanaugh** [comment]
This is well-covered in the [FAQ](https://github.com/Microsoft/TypeScript/wiki/FAQ)

**Comment by Lusito** [comment]
OK, thanks for the information.
